content/post/2020-03-03-packageRank/index.Rmd

```diff
@@ -0,0 +1,854 @@
+  col.ticks = "red", labels = pct.label, cex.axis = 0.8)
+axis(4, at = crosstab[pkg], col.axis = "red", col.ticks = "red",
+  cex.axis = 0.8, labels = format(crosstab[pkg], big.mark = ","))
+points(which(names(crosstab) == pkg), crosstab[pkg], col = "red")
+points(which(names(crosstab) == names(crosstab[1])), y.max,
+  col = "dodgerblue")
+text(which(names(crosstab) == names(crosstab[1])), y.max, pos = 4,
+  labels = paste(names(crosstab[1]), "=", format(crosstab[1],
+  big.mark = ",")), cex = 0.8, col = "dodgerblue")
+text(max(xlim), max(ylim),
+  labels = paste("Tot = ", format(sum(crosstab), big.mark = ",")), cex = 0.8,
+  col = "dodgerblue", pos = 2)
+day <- weekdays(as.Date(date), abbreviate = TRUE)
+title(main = paste0(pkg, " @ ", date, " (", day, ")"))
+```
+
+<br/>
+
+These graphs, customized to be on the same scale, plot the _rank order_ of packages' download counts (x-axis) against the logarithm of those counts (y-axis). It then highlights a package's position in the distribution along with its rank percentile and download count (in red). In the background, the 75th, 50th and 25th percentiles are plotted as dotted vertical lines; the package with the most downloads, which in both cases is [`'magrittr'`](https://cran.r-project.org/package=magrittr) (in blue, top left); and the total number of downloads, 5,561,681 and 3,403,969 respectively (in blue, top right).
+
+<br/>
+
+## limitations
+
+There are (at least) three limitations to `packageRank()`.
+
+### computational
+
+The computational limitation stems from the fact that, unlike `cranlogs::cran_download()`, which benefits from server-side support (i.e., download counts are "pre-computed"), `packageRank()` must download [logs](http://cran-logs.rstudio.com/) over the internet and compute the rank percentiles of download counts. This, however, is somewhat mitigated by caching downloaded log files using the [`'memoise'`](https://CRAN.R-project.org/package=memoise) package,
+
+### analytical
+
+The analytic limitation stems from the computational one. Anything beyond a one-day, cross-sectional comparison (e.g., rank percentiles over time) is "expensive". You need to download all the desired log files, each of which can push 50 MB. If you want to compare ranks for a week, you have to download 7 log files. If you want to compare ranks for a month, you have to download 30 odd log files. As a proof-of-concept of the potential of following this path, the plot below compares nominal download counts with their rank percentiles for [`'cholera'`](https://cran.r-project.org/package=cholera) for the first week in March. Note that, to the chagrin of some, two independently scaled y-variables are plotted on the same graph (gray/black for counts on the left axis, red for rank percentiles on the right).
+
+```{r counts_ranks, fig.align = "center", hugoopts = list(title = "Figure 10 Comparison of Package Download Counts and Rank Percentiles"), echo = FALSE}
```

Comment:
the gray has very low contrast compared to the background https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html